### ISSUE TITLE
ihaskell-hvega: Upper bound on hvega

### ISSUE THREAD

**Issue by lehins** [DougBurke/hvega#95]
Anything prevents `ihaskell-hvega` not to work with `hvega-0.4.x`? Currently `ihaskell-hvega-0.2.0.2` has an upper bound in cabal file `hvega < 0.4`. Is it possible to either make a revision to the cabal file or release the current `ihaskell-hvega-0.2.0.3` on Hackage?

**Comment by DougBurke** [comment]
Input welcome on this.

So, 'hvega-0.4.x' creates v3 of the Vega-Lite spec. Unfortunately, the mimetype that IHaskell recognizes for Vega-Lite is for an older version of the spec, and so you can find that the plots either don't work, or are displayed incorrectly, in IHaskell. This has been address in IHaskell (in that we can now add custom mimetypes - see https://github.com/gibiansky/IHaskell/issues/1089 ) but hasn't been released yet.

So, I've been reluctant to update `ihaskell-hvega` to support `hvega-0.4` as I found it rather demoralizing to create a visualization and then not have it render correctly. Unfortunately the `hvega` API doesn't make it obvious if you are using functionality from v3, v2, or v1 of the spec, so it's hard to tell people "if you need this feature, it may not work in IHaskell".

**Comment by lehins** [comment]
So if I understand correctly, we simply need to wait for a new version of `ipython-kernel` to be released and then the version bound can be relaxed, right? If that's the culprit it would not be unreasonable to ask ihaskell maintainers to cut such new release, considering that it is blocking so much of your being used in ihaskell.

I might be misunderstanding something though, so here is a question from someone very new to VegaLite and `hvega`. In the documentation of both `hvega-0.3.x` and `hvega-0.4.x` it says:
>  targeting version 3 of the JSON schema. 

Does above quote mean that `hvega-0.3.x` simply didn't use the features from v3 spec of VegaLite, that is why plots render correctly with currently released ihaskell (that doesn't yet support new v3 mimes)? 

> I found it rather demoralizing to create a visualization and then not have it render correctly

That is certainly not something users would like to experience, so I definitely understand your reluctance.

**Comment by DougBurke** [comment]
Well, I best point out this this is all theoretical/assumptions in my head; I haven't had a chance to build the new IHaskell and try out the new mimetype support to check it out!

**Comment by lehins** [comment]
The assumptions in your head aren't purely theoretical.
Trying `hvega-0.4.1.1` + `ihaskell-hvega-0.2.0.3` (in the repo) with latest `ipython-kernel` on hackage resulted in:
```
[E 22:42:31.799 LabApp] Notebook JSON is invalid: Additional properties are not allowed ('header' was unexpected)
```
but switching to ipython-haskell from master at https://github.com/gibiansky/IHaskell/tree/2318ee2a90cfc98390651657aec434586b963235
resulted in a correctly rendered plot.

**Comment by DougBurke** [comment]
I've just released ihaskell-hvega-0.2.1.0 which supports hvega 0.4 and the export of VegaLiteLab type. Let me know how it goes.

**Comment by lehins** [comment]
That's great, thank you!
